### PR TITLE
Add Power of Attorney Refresh Button to Case Details

### PR DESCRIPTION
**Pull request by Jeremy1026**
Resolves [CASEFLOW-867](https://vajira.max.gov/browse/CASEFLOW-867)

### Description
Adds a button on to the Case Details page to refresh the Power of Attorney. Shows status of the refresh after clicking the button. Refreshes are limited to once every 10 minutes for each Power of Attorney.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Tests pass
  
### Testing Plan
- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

Refresh Button:

image

Banner showing successful update:

image

Banner showing timeout restrictions are in affect:

image

### CODE REVIEW

File: client/app/queue/reducers.js [hunk 1]
```diff
@@ -172,6 +172,22 @@ const editNodDateUpdates = (state, action) => {
   });
 };
 
+// const poaSyncDateUpdates = (state, action) => {
```
Comment (bgantick) [inline client/app/queue/reducers.js:175]: Remove if not being used

File: client/app/queue/constants.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ export const ACTIONS = {
   RESET_DECISION_OPTIONS: 'RESET_DECISION_OPTIONS',
   EDIT_APPEAL: 'EDIT_APPEAL',
   EDIT_NOD_DATE_UPDATES: 'EDIT_NOD_DATE_UPDATES',
+  // POA_SYNC_DATE_UPDATES: 'POA_SYNC_DATE_UPDATES',
```
Comment (bgantick) [inline client/app/queue/constants.js:41]: Remove if not being used

File: client/app/queue/components/PoaSyncDate.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// import { editAppeal, poaSyncDateUpdates } from '../QueueActions';
```
Comment (bgantick) [inline client/app/queue/components/PoaSyncDate.jsx:3]: Remove if not being used

File: client/app/queue/QueueActions.js [hunk 4]
```diff
@@ -222,6 +222,14 @@ export const editNodDateUpdates = (appealId, nodDateUpdate) => ({
   }
 });
 
+// export const poaSyncDateUpdates = (appealId, poaSyncDate) => ({
```
Comment (bgantick) [inline client/app/queue/QueueActions.js:225]: Remove if not being used

**Review by bgantick — CHANGES_REQUESTED** [review]

**Review by Jeremy1026 — COMMENTED** [review]
Resolved first round of review comments.

**Review by Jeremy1026 — COMMENTED** [review]
Made requested updates.

File: client/app/queue/reducers.js (hunk 1 above)
Comment (Jeremy1026) [reply client/app/queue/reducers.js]: Removed

File: client/app/queue/constants.js (hunk 2 above)
Comment (Jeremy1026) [reply client/app/queue/constants.js]: Removed

File: client/app/queue/components/PoaSyncDate.jsx (hunk 3 above)
Comment (Jeremy1026) [reply client/app/queue/components/PoaSyncDate.jsx]: Removed

File: client/app/queue/QueueActions.js (hunk 4 above)
Comment (Jeremy1026) [reply client/app/queue/QueueActions.js]: Removed

File: app/controllers/appeals_controller.rb [hunk 5]
```diff
@@ -58,14 +58,23 @@ def document_count
...
-      representative_name: appeal.representative_name,
-      representative_address: appeal.representative_address,
-      representative_email_address: appeal.representative_email_address,
-      representative_tz: appeal.representative_tz,
-      poa_last_synced_at: appeal.poa_last_synced_at
-    }
+    render json: power_of_attorney_data
+  end
+
+  def update_power_of_attorney
```
Comment (bgantick) [inline app/controllers/appeals_controller.rb:64]: I'm not seeing logic anywhere to remove a POA if one isn't found via BGS (see https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Bat-Team-Quick-Ref#update-or-delete-bgs-power-of-attorney-poa)

**Review by bgantick — CHANGES_REQUESTED** [review]

File: app/controllers/appeals_controller.rb (hunk 5 above)
Comment (Jeremy1026) [reply app/controllers/appeals_controller.rb:64]: That wasn't part of the original ticket. Might be worth a new ticket, but don't think it should be added to this one, especially not at this point.

Comment (bgantick) [reply app/controllers/appeals_controller.rb:64]: I would consider that part of a "successful" POA refresh

Comment (Jeremy1026) [reply app/controllers/appeals_controller.rb:64]: The deletion when `:not_found` has been added, including a test for this situation.

File: client/app/queue/components/PoaRefreshButton.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        {buttonText}
+      </Button>
+    </div>
+    )}
+  </React.Fragment>;
+};
+
+PoaRefreshButton.propTypes = {
+  powerOfAttorney: PropTypes.shape({
+    poa_last_synced_at: PropTypes.string,
```
Comment (ajspotts) [inline client/app/queue/components/PoaRefreshButton.jsx:45]: don't believe we need this propType in this file

File: client/app/queue/PowerOfAttorneyDetail.jsx [hunk 7]
```diff
@@ -30,7 +31,7 @@ const powerOfAttorneyFromAppealSelector = (appealId) =>
     return {
       powerOfAttorney: appeal?.powerOfAttorney,
       loading: false,
-      error: loadingPowerOfAttorney?.error
+      error: loadingPowerOfAttorney?.error,
```
Comment (ajspotts) [inline client/app/queue/PowerOfAttorneyDetail.jsx:34]: Is this trailing comma necessary?

File: client/app/queue/components/PoaRefreshButton.jsx (hunk 6 above)
Comment (Jeremy1026) [reply client/app/queue/components/PoaRefreshButton.jsx]: Resolved in e9e4867

File: client/app/queue/PowerOfAttorneyDetail.jsx (hunk 7 above)
Comment (Jeremy1026) [reply client/app/queue/PowerOfAttorneyDetail.jsx]: Removed in 9d83b3c

**Review by ajspotts — APPROVED** [review]
LGTM! 🎉

**Pull request merged**